explainer.md

```diff
@@ -405,20 +393,67 @@ function beginInlineXRSession() {
+
+Features may be unavailable for a number of reasons, among which is the fact not all devices which support WebXR can support the full set of features. Another consideration is that some features expose [sensitive information](privacy-security-explainer.md#sensitive-information) which may require a clear signal of [user intent](privacy-security-explainer.md#user-intent) before functioning. Any feature which requires this signal to be provided via [explicit consent](privacy-security-explainer.md#explicit-consent) must request this consent prior to the session being created. This ensures a consistent experience across all hardware form-factors, regardless of whether the UA has a [trusted immersive UI](privacy-security-explainer.md#trusted-immersive-ui) available.
+
+WebXR allows the following features to be requested:
+* `local`
+* `local-floor`
+* `bounded-floor`
+* `unbounded`
+
+This list is currently limited to a subset of reference space types, but in the future will expand to include additional features. Some potential future features under discussion that would be candidates for this list are: eye tracking, plane detection, geo alignment, etc.
+
+Developers communicate their feature requirements by categorizing them into one of the following groups:
+* **Required** This feature must be available in order for the experience to function at all. If [explicit consent](privacy-security-explainer.md#explicit-consent) is necessary, users will be prompted in response to `xr.requestSession()`. Session creation will be rejected if the feature is unavailable for the XR device or if the UA determines the user does not wish the feature enabled.
+* **Optional** The experience would like to use this feature for the entire session, but can function without it. Again, if [explicit consent](privacy-security-explainer.md#explicit-consent) is necessary, users will be prompted in response to `xr.requestSession()`. However, session creation will succeed regardless of the feature's hardware support or user intent. Developers must not assume optional features are available in the session and check the result from attempting to use them.
+
+Features recognized by the UA that are not explicitly listed in these arrays will be implicitly added to `XRSessionInit.optionalFeatures` if they do not require consent or [consent has been implied](privacy-security-explainer.md#implied-consent). For example, creating an `immersive-vr` session implies consent for `local` and `local-floor` reference spaces even if not explicitly listed in `XRSessionInit`. If the UA does not recognize an entry in `XRSession.requiredFeatures` the session will not be created.
```

Comment:
> Features recognized by the UA that are not explicitly listed in these arrays will be implicitly added to `XRSessionInit.optionalFeatures` if they do not require consent or consent has been implied.

I'm sure how to interpret this line. Are you stating that if user agent A that has determined it can safely offer, say, eye-tracking without user consent should implicitly append it to the optional array? (*Terrible* example, I know, but work with me.) That seems problematic, because then if user agent B decides that eye-tracking really *does* require explicit consent, experiences written only against A will fail because the feature was not requested.

I'm onboard with `local` and `local-floor` being implicit for `immersive-vr`, but I feel like they probably need to be part of an explicit list of implied features rather than something left to the user agent.

Reply:
>I'm sure how to interpret this line. Are you stating that if user agent A that has determined it can safely offer, say, eye-tracking without user consent should implicitly append it to the optional array? (Terrible example, I know, but work with me.) That seems problematic, because then if user agent B decides that eye-tracking really does require explicit consent, experiences written only against A will fail because the feature was not requested.

Perhaps this can be fixed by changing the sentence to say "if they do not require a signal of user intent"?  For context, I added this sentence based on our noodling about the idea that maybe there are features that have no consent requirements at all, such as the ability to support an xr-standard gamepad.  At the time you were hesitant to include it because if gamepad based motion controllers are present, they should always appear in the inputSources array.  My case for including it on the list was that there are some experiences which will not work if the hardware cannot support a multibutton input source.  I tried to strike the balance by suggesting only automatically enabling features that are normatively specified to not be dependent on a signal of user intent.  Does that make more sense?

Reply:
> I'm onboard with local and local-floor being implicit for immersive-vr, but I feel like they probably need to be part of an explicit list of implied features rather than something left to the user agent.

My next PR calls this out a lot more concretely as it digs into threats and mitigations, but I'm ok with making some sort of a list here if it helps.  I only left it out of here because the list is already located in the [Spatial tracking explainer](spatial-tracking-explainer.md#xrreferencespace-availability)

Reply:
Perhaps both issues can be address with a single list of "exceptions" added to this section?

Reply:
> Perhaps this can be fixed by changing the sentence to say "if they do not require a signal of user intent"?

Ah, I'm see what you were trying to say. I'm OK with this mental model of "everything that doesn't require user intent is automatically optional". It does help clarify the gamepad case, and I'm sure others. However, we'd still want an explicit list of which features those are so that we don't get UA disagreements.

Reply:
Ok, how about I switch the whole paragraph around to this:

> If the UA does not recognize an entry in `XRSession.requiredFeatures` the session will not be created. Under some circumstances, features recognized by the UA that are not explicitly listed in these arrays may be implicitly added to `XRSessionInit.optionalFeatures`. For example, if a feature does not require a signal of [user intent](privacy-security-explainer.md#user-intent). Or, if the requested session mode implies consent, such as `immersive-vr` does for `local` and `local-floor` reference spaces.

Reply:
Better, thanks!